src/mscorlib/shared/System/Collections/Generic/EmptyEnumerator.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Collections.Generic
+{
+    internal class EmptyEnumerator<T> : IEnumerator<T>, IDictionaryEnumerator
+    {
+        public static EmptyEnumerator<T> Shared { get; } = new EmptyEnumerator<T>();
+
+        private EmptyEnumerator() { }
+
+        public T Current { get; } = default;
+
+        public bool MoveNext() => false;
+
+        public void Dispose() { }
+
+        DictionaryEntry IDictionaryEnumerator.Entry { get; } = default;
+
+        object IDictionaryEnumerator.Key { get; } = null;
+
+        object IDictionaryEnumerator.Value { get; } = null;
+
+        object IEnumerator.Current { get; } = null;
```

Comment:
Why are all these properties autoproperties that load their value from a field, when they could just directly return it? It's going to result in shorter source code and it should also be more efficient (though probably not significantly, unless the CLR gets better at devirtualization).

E.g. change the last one to:

```c#
object IEnumerator.Current => null;
```

Reply:
Fixed